pages/blocks/headerwithlockup/headerwithlockup.js

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+const header = document.querySelector('.headerwithlockup div');
+let hasText = null;
+let hasNav = null;
+
+if (header.querySelector('div:first-of-type').childElementCount > 1) {
```

Comment:
you may be missing a `:scope` on the selector, this query might not do exactly what you think it does.. I could definitely be wrong though: https://developer.mozilla.org/en-US/docs/Web/API/Element/querySelector#the_entire_hierarchy_counts

also more generally.. it's good to query once then check the variable, especially when using variables that may be `null` or `undefined` like the return from `querySelector()`
```js
const div = header.querySelector('div:first-of-type');
if (div && div.childElementCount > 1) {
  div.classList.add('has-text');
  hasText = true;
}
```

Reply:
nit: not sure what `has-text` is suggesting, but a node that doesn't have a child element *can* have innerText

Reply:
Oh awesome. Thanks for these comments. I made changes based on the feedback!